File: CWE_JavaScipt/src/CWE306/Complaint.js
//      CWE-306	Missing Authentication for Critical Function Prevention

class User {

    setAccountNumber(accNo) {
            this.accountNumber = accNo;
        }

    setAccountType(accType) {
            this.accountType = accType;
        }

    setAccountOwnerName(accName) {
            this.accountName = accName;
        }

    setBalance(bal) {
            this.balance = bal;
        }

    getAccountDetail(){
    console.log("AccountNumber:", this.accountNumber)
    console.log("Owner Name:", this.accountName)


    }

}

function createBankAccount(authKey,accNo,accType,accOwner,balance) {
   const user = new User();
if(authKey=="key"){
   user.setAccountNumber(accNo);
   user.setAccountType(accType);
   user.setAccountOwnerName(accOwner);
   user.setBalance(balance);


}
 return user;
}

const user1 = createBankAccount("authkey","123","saving","Owner",300)

user1.getAccountDetail()

File: CWE_Java/src/main/java/com/example/cwe/CWE306/Complaint/Complaint.java
package com.example.cwe.CWE306.Complaint;

//      CWE-306: Missing Authentication for Critical Function

import java.util.*;

public class Complaint {
    public static void main(String[] args) {

        BankAccount user1 = createBankAccount("authKey","123","saving","Owner",300);
        user1.getOwnerName();
    }

    public static BankAccount createBankAccount(String authenticationKey,String accountNumber,
                                                String accountType, String accountOwnerName, double balance) {
        BankAccount account = new BankAccount();
        if (Objects.equals(authenticationKey, "authKey")) {
            account.setAccountNumber(accountNumber);
            account.setAccountType(accountType);
            account.setAccountOwnerName(accountOwnerName);
            account.setBalance(balance);
        }
        return account;
    }
}
